x/mongo/driver/topology/connection_test.go

```diff
@@ -485,6 +485,41 @@ func TestConnection(t *testing.T) {
 					}
 					listener.assertMethodsCalled(t, 1, 1)
 				})
+				t.Run("message too large errors", func(t *testing.T) {
+					testCases := []struct {
+						name   string
+						buffer []byte
+						desc   description.Server
+					}{
+						{
+							"message too large errors with small max message size",
+							[]byte{0x0A, 0x00, 0x00, 0x00, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A},
+							description.Server{MaxMessageSize: 9},
+						},
+						{
+							"message too large errors with default max message size",
+							append([]byte{0x04, 0x6C, 0xDC, 0x02}, make([]byte, 4800000)...),
```

Comment:
If the code is supposed to error out after only reading the first 4 bytes, you shouldn't need to fill in the rest of the slice with 48MB of data. Just the first 4 should be sufficient. I'd also recommend adding a comment on the side with the length of the message because it's not immediately obvious what {0x04, 0x6c, 0xdc, 0x02} is supposed to be.

Reply:
Yeah that's a good point; I'll add the comment. Thought it'd be good to mimic the full data, but that's definitely true, so I can just remove the `make(...)`.